crates/solver/src/settlement_simulation.rs

```diff
@@ -48,41 +54,47 @@ pub async fn simulate_and_estimate_gas_at_current_block(
         return Ok(Vec::new());
     }
 
-    let web3 = web3::Web3::new(web3::transports::Batch::new(web3.transport().clone()));
+    let web3 = web3::Web3::new(shared::transport::buffered::Buffered::new(
+        web3.transport().clone(),
+    ));
+    let contract_with_buffered_transport = GPv2Settlement::at(&web3, contract.address());
     let mut results = Vec::new();
     for chunk in settlements.chunks(SIMULATE_BATCH_SIZE) {
-        let calls = chunk
+        let txs = chunk
             .iter()
             .map(|(account, settlement)| {
-                let tx = settle_method(gas_price, contract, settlement.clone(), account.clone()).tx;
-                let resolved_gas_price = tx
-                    .gas_price
-                    .map(|gas_price| gas_price.resolve_for_transaction())
-                    .unwrap_or_default();
-                let call_request = CallRequest {
-                    from: tx.from.map(|account| account.address()),
-                    to: tx.to,
-                    gas: None,
-                    gas_price: resolved_gas_price.gas_price,
-                    value: tx.value,
-                    data: tx.data,
-                    transaction_type: resolved_gas_price.transaction_type,
-                    access_list: None,
-                    max_fee_per_gas: resolved_gas_price.max_fee_per_gas,
-                    max_priority_fee_per_gas: resolved_gas_price.max_priority_fee_per_gas,
+                settle_method(
+                    gas_price,
+                    &contract_with_buffered_transport,
+                    settlement.clone(),
+                    account.clone(),
+                )
+                .tx
+            })
+            .collect::<Vec<_>>();
+        let mut access_lists = access_list_estimator
+            .estimate_access_lists(txs.as_slice())
+            .await
+            .unwrap_or_default()
+            .into_iter();
+        let calls = txs
+            .into_iter()
+            .map(|tx| {
+                let tx = match access_lists.next().unwrap() {
```

Comment:
I believe `unwrap()` is able to panic here if `estimate_access_lists()` doesn't return an access list for every transaction because some bogous acess lists got filtered out.

Also if `estimate_access_lists()` throws an error `unwrap_or_default()` would return an empty `Vec` resulting in the same problem.

Reply:
Changed the logic a bit.